src/main/java/io/lettuce/core/RedisClient.java

```diff
@@ -221,46 +222,83 @@ public static RedisClient create(ClientResources clientResources, RedisURI redis
 
-        return getConnection(connectStandaloneAsync(newStringStringCodec(), redisURI, redisURI.getTimeout()));
+        return getConnection(connectStandaloneAsync(newStringStringCodec(), redisURI, redisURI.getTimeout(), null));
     }
 
     /**
      * Open a new connection to a Redis server using the supplied {@link RedisURI} and the supplied {@link RedisCodec codec} to
      * encode/decode keys.
      *
-     * @param codec Use this codec to encode/decode keys and values, must not be {@code null}
+     * @param codec    Use this codec to encode/decode keys and values, must not be {@code null}
      * @param redisURI the Redis server to connect to, must not be {@code null}
-     * @param <K> Key type
-     * @param <V> Value type
+     * @param <K>      Key type
+     * @param <V>      Value type
      * @return A new connection
      */
     public <K, V> StatefulRedisConnection<K, V> connect(RedisCodec<K, V> codec, RedisURI redisURI) {
 
         assertNotNull(redisURI);
 
-        return getConnection(connectStandaloneAsync(codec, redisURI, redisURI.getTimeout()));
+        return getConnection(connectStandaloneAsync(codec, redisURI, redisURI.getTimeout(), null));
     }
 
     /**
      * Open asynchronously a new connection to a Redis server using the supplied {@link RedisURI} and the supplied
      * {@link RedisCodec codec} to encode/decode keys.
      *
-     * @param codec Use this codec to encode/decode keys and values, must not be {@code null}
+     * @param codec    Use this codec to encode/decode keys and values, must not be {@code null}
      * @param redisURI the Redis server to connect to, must not be {@code null}
-     * @param <K> Key type
-     * @param <V> Value type
+     * @param <K>      Key type
+     * @param <V>      Value type
      * @return {@link ConnectionFuture} to indicate success or failure to connect.
      * @since 5.0
      */
     public <K, V> ConnectionFuture<StatefulRedisConnection<K, V>> connectAsync(RedisCodec<K, V> codec, RedisURI redisURI) {
 
         assertNotNull(redisURI);
 
-        return transformAsyncConnectionException(connectStandaloneAsync(codec, redisURI, redisURI.getTimeout()));
+        return transformAsyncConnectionException(connectStandaloneAsync(codec, redisURI, redisURI.getTimeout(), null));
+    }
+
+
+    /**
+     * Open a new connection to a Redis server using the supplied {@link RedisURI} and the supplied {@link RedisCodec codec} to
+     * encode/decode keys.
+     *
+     * @param codec     Use this codec to encode/decode keys and values, must not be {@code null}
+     * @param redisURI  the Redis server to connect to, must not be {@code null}
+     * @param listeners redis command listeners
+     * @param <K>       Key type
+     * @param <V>       Value type
+     * @return A new connection
+     */
+    public <K, V> StatefulRedisConnection<K, V> connect(RedisCodec<K, V> codec, RedisURI redisURI, List<RedisCommandListener> listeners) {
```

Comment:
Let's keep listeners as an intrinsic property that should not be passed to the `connect` method. I think `ClientResources` is a good place to hold the configuration.

Reply:
Totally agree. Fixed